tests/lib/cmsis_dsp/matrix/src/main.c

```diff
@@ -1,63 +1,21 @@
 /*
- * Copyright (c) 2021 Stephanos Ioannidis <root@stephanos.io>
+ * Copyright (c) 2020 Stephanos Ioannidis <root@stephanos.io>
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #include <zephyr/ztest.h>
-#include <zephyr/kernel.h>
 
-extern void test_matrix_unary_q7(void);
-extern void test_matrix_unary_q15(void);
-extern void test_matrix_unary_q31(void);
-extern void test_matrix_unary_f16(void);
-extern void test_matrix_unary_f32(void);
-extern void test_matrix_unary_f64(void);
-
-extern void test_matrix_binary_q7(void);
-extern void test_matrix_binary_q15(void);
-extern void test_matrix_binary_q31(void);
-extern void test_matrix_binary_f16(void);
-extern void test_matrix_binary_f32(void);
-extern void test_matrix_binary_f64(void);
-
-void test_main(void)
-{
-#ifdef CONFIG_CMSIS_DSP_TEST_MATRIX_UNARY_Q7
-	test_matrix_unary_q7();
-#endif
-#ifdef CONFIG_CMSIS_DSP_TEST_MATRIX_UNARY_Q15
-	test_matrix_unary_q15();
-#endif
-#ifdef CONFIG_CMSIS_DSP_TEST_MATRIX_UNARY_Q31
-	test_matrix_unary_q31();
-#endif
-#ifdef CONFIG_CMSIS_DSP_TEST_MATRIX_UNARY_F16
-	test_matrix_unary_f16();
-#endif
-#ifdef CONFIG_CMSIS_DSP_TEST_MATRIX_UNARY_F32
-	test_matrix_unary_f32();
-#endif
-#ifdef CONFIG_CMSIS_DSP_TEST_MATRIX_UNARY_F64
-	test_matrix_unary_f64();
-#endif
+/* This is a empty testsuite when no config
+ * can be built.
+ */
+#ifndef CONFIG_CMSIS_DSP_FLOAT16
 
-#ifdef CONFIG_CMSIS_DSP_TEST_MATRIX_BINARY_Q7
-	test_matrix_binary_q7();
-#endif
-#ifdef CONFIG_CMSIS_DSP_TEST_MATRIX_BINARY_Q15
-	test_matrix_binary_q15();
-#endif
-#ifdef CONFIG_CMSIS_DSP_TEST_MATRIX_BINARY_Q31
-	test_matrix_binary_q31();
-#endif
 #ifdef CONFIG_CMSIS_DSP_TEST_MATRIX_BINARY_F16
-	test_matrix_binary_f16();
+ZTEST_SUITE(matrix_binary_f16, NULL, NULL, NULL, NULL, NULL);
```

Comment:
Can you explain why this is needed?

If this is because the F16 tests (e.g. `libraries.cmsis_dsp.matrix.unary_f16`) run with `CONFIG_CMSIS_DSP_TEST_MATRIX_UNARY_F16=n` because of `depends on CMSIS_DSP_FLOAT16`, then the `filter:` in the `testcase.yaml` should be updated to filter by `CONFIG_CMSIS_DSP_FLOAT16` (note that it is `default y` when FP16 is supported).

Reply:
Yes, it is. And I follow your suggestion to put  CONFIG_CMSIS_DSP_FLOAT16 in the filter testcase.yaml, it is better now and I can keep the dependency of  CMSIS_DSP_FLOAT16 you mentioned. Thanks so much for the review, lots of files. :)